### PR TITLE
Extend release-workflow documentation

### PR DESCRIPTION
**Pull request by mfussenegger**
First commit is a bit of a restructuring of DEVELOP.rst.
The second commit contains the workflow documentation. I deliberatly kept it a
bit unspecific in terms of when things happen and how long they last to ensure
the documentation doesn't get outdated too quickly

### CODE REVIEW

File: devs/docs/release.rst [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+focus on fixes, performance improvements or other cleanups.
+
+
+At some point we'll lift the feature freeze. At this point a ``major.minor``
+release branch is created and we bump the ``minor`` version on master.
+
+
+When the number of bugs discovered in the release candidates decreases enough
+and it's considered stable, we move the release candidate into our stable
+repositories and declare it stable.
```
Comment (joemoe) [inline devs/docs/release.rst]: would add something like `so this will be the first stable release of this ``minor`` version.

**Comment by mfussenegger** [comment]
@joemoe @chaudum can you take another look?

File: devs/docs/basics.rst [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    $ ./gradlew distTar
+
+The tarball can then be found in the ``app/build/distributions`` directory.
+
+To get a full list of all available tasks, run::
+
+    $ ./gradlew tasks
+
+
+Running tests
```
Comment (chaudum) [inline devs/docs/basics.rst]: Tests

File: devs/docs/release.rst [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+======================
+
+- Development happens in feature branches. Everything that goes into master
+  must be reviewed.
+
+- PRs should be small but self-contained. "Dead code" (Features that aren't
+  integrated) cannot go into master.
+
+- Commits must be self-contained. Each commit must compile and tests should
+  pass. (Flaky tests happen.)
```
Comment (mxm) [inline devs/docs/release.rst]: "Flaky tests happen" sounds like they are acceptable. How about "Beware of Flaky tests"?

File: devs/docs/release.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+- Create a tag by running ``./devs/tools/create_tag.sh``
+
+- Create a "version bump" commit. (See workflow description above)
+
+You can build a release tarball like so::
+
+    $ ./gradlew release
+
+This task runs the ``distTar`` task but also checks that the output of ``git
+describe --tag`` matches the current version of CrateDB.
```
Comment (mxm) [inline devs/docs/release.rst]: Not sure this paragraph is needed.

File: devs/docs/tests.rst [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    $ ./gradlew -PtestLogging -Dtests.loggers.levels=io.crate:DEBUG,io.crate.planner.consumer.NestedLoopConsumer:TRACE :sql:test
+
+
+Or with code changes:
+
+Use ``@TestLogging(["<packageName1>:<logLevel1>", ...])`` on your test class or
+test method to enable more detailed logging. For example::
+
+    @TestLogging("io.crate:DEBUG,io.crate.planner.consumer.NestedLoopConsumer:TRACE")
```
Comment (mxm) [inline devs/docs/tests.rst]: 👍 Very helpful for new contributors.

**Pull request merged**